infrastructure/src/main/java/org/corfudb/infrastructure/log/InMemoryStreamLog.java

```diff
@@ -124,6 +125,25 @@ private void throwLogUnitExceptionsIfNecessary(long address, LogData entry) {
         }
     }
 
+    /**
+     * Returns the known addresses in this Log Unit in the specified consecutive
+     * range of addresses.
+     *
+     * @param range Range of addresses to compare.
+     * @return Set of known addresses.
+     */
+    @Override
+    public Set<Long> getKnownAddressesInRange(Range<Long> range) {
```

Comment:
This API is actually not very clear, because the Range can be any combination of open/close endpoints, but the for loop on like 139 assumes that the range is closed/closed. Essentially making an assumption about a generic. For example, [1,2,3) would can return 1,2 and 3, but based on the interface it should return 1 and 2

Reply:
Maybe just use the endpoints directly, no need to wrap them in a Range ?